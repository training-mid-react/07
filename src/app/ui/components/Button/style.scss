
@use '../../styles/variables' as var;

.custom-button {
    background-color: var.$color-green;
    border-radius: 5px;
    border: none;
    color: var.$color-white;
    cursor: pointer;
    font-weight: bold;
    padding: 12px 20px;
    transition: background-color 0.3s ease;
  
    &:hover {
      background-color: darken(var.$color-dark-green, 10%);
    }
  
    &:disabled {
      background-color: var.$color-gray;
      cursor: not-allowed;
    }
  }
  