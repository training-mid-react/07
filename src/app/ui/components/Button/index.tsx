import React from 'react';
import './style.scss';

interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, disabled, className }) => {
  return (
    <button data-testid='button-component' className={className || 'custom-button'} onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;
