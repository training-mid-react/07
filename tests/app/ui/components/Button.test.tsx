import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../../../../src/app/ui/components/Button';

describe('Componente Button', () => {
    it('renderiza correctamente', () => {
      const { asFragment } = render(<Button onClick={() => {}}>Click me</Button>);
      expect(asFragment()).toMatchSnapshot();
    });
  
    it('renderiza con className', () => {
      const { asFragment } = render(<Button className="custom-class" onClick={() => {}}>Click me</Button>);
      expect(asFragment()).toMatchSnapshot();
    });
  
    it('renderiza con el estado deshabilitado', () => {
      const { asFragment } = render(
        <Button onClick={() => {}} disabled>
          Click me
        </Button>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  
    it('dispara la función onClick cuando se hace click', () => {
      const mockOnClick = vi.fn();
      const { getByTestId } = render(<Button onClick={mockOnClick}>Click me</Button>);
      const button = getByTestId('button-component');
      fireEvent.click(button);
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  
    it('no dispara onClick cuando está deshabilitado', () => {
      const mockOnClick = vi.fn();
      const { getByTestId } = render(
        <Button onClick={mockOnClick} disabled>
          Click me
        </Button>
      );
      const button = getByTestId('button-component');
      fireEvent.click(button);
      expect(mockOnClick).not.toHaveBeenCalled();
    });
  });