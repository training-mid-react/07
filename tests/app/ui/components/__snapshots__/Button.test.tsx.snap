// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Button > renderiza con className 1`] = `
<DocumentFragment>
  <button
    class="custom-class"
    data-testid="button-component"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Componente Button > renderiza con el estado deshabilitado 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
    data-testid="button-component"
    disabled=""
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Componente Button > renderiza correctamente 1`] = `
<DocumentFragment>
  <button
    class="custom-button"
    data-testid="button-component"
  >
    Click me
  </button>
</DocumentFragment>
`;
